### PR TITLE
Fixed StrictMode DiskReadViolation and DiskWriteViolation

### PR DESCRIPTION
**Pull request by Jessica-Kuo**
File containing logcat of the StrictMode error: (starts at line 152) 
[StrictMode_Logcat_before.pdf](https://github.com/segmentio/analytics-android/files/5718129/StrictMode_Logcat_before.pdf)

File containing logcat with the fix:
[StrictMode_Logcat_after.pdf](https://github.com/segmentio/analytics-android/files/5718133/StrictMode_Logcat_after.pdf)

### CODE REVIEW

**Comment by prayansh** [comment]
Since the violation only happens on our own SegmentIntegration can we narrow down the scope of the thread here? I would suggest somewhere here instead: https://github.com/segmentio/analytics-android/blob/4555626794d4e63c8a1f9d98e54eca86c97f262e/analytics/src/main/java/com/segment/analytics/SegmentIntegration.java#L185

**Comment by Jessica-Kuo** [comment]
Updated! New logcat file: 
[StrictMode_after_fix.pdf](https://github.com/segmentio/analytics-android/files/5718286/StrictMode_after_fix.pdf)

File: analytics/src/main/java/com/segment/analytics/SegmentIntegration.java [hunk 1]
```diff
@@ -182,22 +183,36 @@ static synchronized SegmentIntegration create(
             Logger logger,
             Crypto crypto,
             ValueMap settings) {
-        PayloadQueue payloadQueue;
+        final PayloadQueue[] payloadQueue = new PayloadQueue[1];
+        Thread execThread =
+                new Thread(
+                        () -> {
+                            try {
+                                File folder =
```
Comment (prayansh) [inline analytics/src/main/java/com/segment/analytics/SegmentIntegration.java:191]: This seems like a great candidate to use Futures. Futures allow you to return values from a thread. I would suggest using those here

**Review by prayansh — CHANGES_REQUESTED** [review]

File: analytics/src/main/java/com/segment/analytics/SegmentIntegration.java (hunk 1 above)
Comment (prayansh) [reply analytics/src/main/java/com/segment/analytics/SegmentIntegration.java:191]: also i just wanted to confirm, but are there are any potential downfalls to creating this file from a random thread? I cant think of any but definitely something to think about

Comment (Jessica-Kuo) [reply analytics/src/main/java/com/segment/analytics/SegmentIntegration.java:191]: When I was making it, I didn't think there were any, but I also didn't look into the potential issues of isolating the process to it's own thread... from what I remember in school, as long as there isn't any other process in the thread it shouldn't deadlock (without mutexes or anything to manage it) so I think its okay?

**Pull request closed without merging**